src/main/kotlin/gg/essential/elementa/components/ScrollComponent.kt

```diff
@@ -469,7 +474,8 @@ class ScrollComponent constructor(
         if (isHorizontal) {
             component.setWidth(RelativeConstraint(clampedPercentage))
         } else {
-            component.setHeight(RelativeConstraint(clampedPercentage))
+            val heightConstraint = RelativeConstraint(clampedPercentage)
+            component.setHeight(if (limitScrollBarGripHeight) ScrollBarGripMinHeightConstraint(heightConstraint) else heightConstraint)
         }
```

Comment:
While we are at it, we should totally implement this for horizontal bars as well. The usability concern 100% exists there as well.